articles/142_idl.md

```diff
@@ -35,3 +35,55 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Type Mapping
+
+The next table defines how IDL types are mapped to the following programming languages:
+
+* C11 (or higher)
+* C++11 (or higher)
+* Python 3 (or higher)
+
+| IDL type    | C type      | C++ type    | Python type          |
+| --------    | ----------- | ----------- | -------------------- |
+| float       | float       | float       | float                |
+| double      | double      | double      | float                |
+| long double | long double | long double | float                |
+| char        | signed char | char        | str with length 1    |
+| wchar       | char16\_t   | char16\_t   | str with length 1    |
+| bool        | \_Bool      | bool        | bool                 |
+| octet       | uint8\_t    | uint8\_t    | bytes with length 1  |
+| int8        | int8\_t     | int8\_t     | int                  |
+| uint8       | uint8\_t    | uint8\_t    | int                  |
+| int16       | int16\_t    | int16\_t    | int                  |
+| uint16      | uint16\_t   | uint16\_t   | int                  |
+| int32       | int32\_t    | int32\_t    | int                  |
+| uint32      | uint32\_t   | uint32\_t   | int                  |
+| int64       | int64\_t    | int64\_t    | int                  |
+| uint64      | uint64\_t   | uint64\_t   | int                  |
+
+The next table defines how IDL templated and constructed types are mapped to the programming languages (unless specified otherwise below).
+
+| IDL type       | C type                 | C++ type            | Python type |
+| -------------- | ---------------------- | ------------------- | ----------- |
+| T[N]           | T[N]                   | std::array<T, N>    | list        |
+| sequence<T>    | struct {size\_t, T * } | std::vector<T>      | list        |
+| sequence<T, N> | struct {size\_t, T * } | custom\_class<T, N> | list        |
+| string         | char *                 | std::string         | str         |
+| string<N>      | char *                 | std::string         | str         |
+| wstring        | char16\_t *            | std::u16string      | str         |
+| wstring<N>     | char16\_t *            | std::u16string      | str         |
+
+These array and sequence types have special mappings.
+If a cell is blank then the default mapping is used.
+
+| IDL type                     | C type                 | C++ type       | Python type |
+| ---------------------------- | ---------------------- | -------------- | ----------- |
```

Comment:
Do we want to specify the mapping to use numpy for scalar type?

Reply:
+1 for `octet[N]` being `np.array()`. It would make it easier to support multiple dimension arrays in the future. numpy seems like a minimal enough dependency. `pip freeze` after installing numpy in a venv only shows `numpy` and `pkg-resources` (which was in the env by default).

Reply:
How about other scalar types and sequences?

Reply:
> +1 for octet[N] being np.array()

For `octet[N]` `bytes` already provides a native mapping (without the usual Python overhead) so I am not sure about that case.